### PR TITLE
added model layer

### PR DESCRIPTION
**Pull request by AliShan302**
This pull request contains model layer

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module server
```
Comment (shahzadhaider1) [inline go.mod]: the module should be the github repo link

like this: github.com/AliShan302/Task_Schedular

File: models/task.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package models
+
+import (
+	"time"
+
+	"github.com/fatih/structs"
+)
+
+type Task struct {
```
Comment (shahzadhaider1) [inline models/task.go:10]: exported type should have a comment

File: models/task.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	Name        string    `json:"name"`
+	CreatedAt   time.Time `json:"created_at"`
+	Status      string    `json:"status"`
+	Description string    `json:"description"`
+	Deadline    time.Time `json:"deadline"`
+	Priority    int       `json:"priority"`
+	Assignee    string    `json:"assignee"`
+}
+
+func (t *Task) Map() map[string]interface{} {
```
Comment (shahzadhaider1) [inline models/task.go:22]: exported type should have a comment

File: models/task.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	return structs.Map(t)
+}
+
+func (t *Task) Names() []string {
```
Comment (shahzadhaider1) [inline models/task.go:27]: exported type should have a comment

File: models/task.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	names := make([]string, len(fields))
+	for i, field := range fields {
+		name := field.Name()
+		tagName := field.Tag(structs.DefaultTagName)
+		if tagName != "" {
+			name = tagName
+		}
+		names[i] = name
+	}
+	return names
```
Comment (shahzadhaider1) [inline models/task.go:39]: new line before return

File: models/task_test.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		"Priority":    2,
+		"Assignee":    "Ali",
+	}
+
+	taskMap := task.Map()
+
+	if !reflect.DeepEqual(taskMap, expectedMap) {
+		t.Errorf("Map() returned unexpected map.\nExpected: %+v\nGot: %+v", expectedMap, taskMap)
+	}
+}
```
Comment (shahzadhaider1) [inline models/task_test.go:94]: tests should have multiple cases including scenarios like, Task with wrong data, not getting what's expected, etc. and any other edge cases you can think of

**Review by shahzadhaider1 — CHANGES_REQUESTED** [review]

**Review by shahzadhaider1 — APPROVED** [review]
Looks good.
